source/Octopus.Tentacle/Communications/AutofacServiceFactory.cs

```diff
@@ -17,48 +17,48 @@ public class AutofacServiceFactory : IServiceFactory, IServiceRegistration, IDis
         // Must never be modified as it is required for backwards compatability in BackwardsCompatibleCapabilitiesV2Decorator
         const string TentacleServiceShuttingDownMessage = "The Tentacle service is shutting down and cannot process this request.";
         readonly ILifetimeScope scope;
-        readonly Dictionary<string, Type> serviceTypes = new();
+        readonly Dictionary<string, KnownService> knownServices = new();
 
         public AutofacServiceFactory(ILifetimeScope scope, IEnumerable<IAutofacServiceSource> sources)
         {
             this.scope = scope.BeginLifetimeScope(b =>
             {
-                foreach (var service in sources.SelectMany(x => x.ServiceTypes.EmptyIfNull()))
+                foreach (var knownService in sources.SelectMany(x => x.KnownServices.EmptyIfNull()))
                 {
-                    BuildService(b, service);
+                    BuildService(b, knownService);
                 }
             });
         }
 
-        void BuildService(ContainerBuilder builder, Type serviceType)
+        void BuildService(ContainerBuilder builder, KnownService knownService)
         {
-            var registrationBuilder = builder
-                .RegisterType(serviceType)
+            //register the service implementation AsSelf(), so we can resolve it in the future
+            builder
+                .RegisterType(knownService.ServiceImplementationType)
                 .AsSelf()
                 .SingleInstance();
 
-            var interfaces = serviceType.GetInterfaces();
-            if (serviceType.IsInterface || interfaces.IsNullOrEmpty())
-            {
-                throw new InvalidServiceTypeException(serviceType);
-            }
-            
-            foreach (var face in interfaces)
```

Comment:
Yes, I saw this once before, how it's extra complication to allow one service to have multiple contracts (which we don't use). So I'm happy to see this unrequired loop go 😁